vault/cors.go

```diff
@@ -0,0 +1,112 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var responseHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
```

Comment:
It absolutely can be a `*` instead of a list. I just was err'ing on the side of being explicit. I'll refactor.